src/assignments/scrabble-scorer.rst

```diff
@@ -0,0 +1,336 @@
+      letter z:  10
+
+Do not panic! Instead of tediously hard-coding ``newPointStructure``, use your
+clever coding skills to create the new object.
+
+
+Code the Function   
+~~~~~~~~~~~~~~~~~~
+
+#. Write a ``Transform`` method that returns a dictionary and has no parameters. Calling
+   ``Transform()`` will *return* a dictionary with *lowercase*
+   letters as keys. The value for each key will be the points assigned to that
+   letter.
+#. Initialize an empty ``newPointStructure`` dictionary by setting it equal to
+   ``Transform()``.
+#. Points to Consider:
+
+   #. Recall that ``foreach`` loops 
+      :ref:`iterate over the key/value pairs <kvp-foreach>` within an dictionary.
+   #. If you need a reminder of how to assign new key/value pairs, review the
+      :ref:`relevant section <initialize-dictionary>` in the
+      ``Control Flow and Collections`` chapter.
+   #. To access the string of letters within ``oldPointStructure``, use bracket
+      notation (``oldPointStructure[key]``).
+   #. To access a particular element within a string, 
+      consider what :ref:`strings are made of <string-chars>` and 
+      the various :ref:`string methods <string-method-examples>` you can use to access them.
+   #. What about the data types.  Will you need to `convert <https://docs.microsoft.com/en-us/dotnet/api/system.convert.tochar?view=net-5.0>`_ or `parse <https://docs.microsoft.com/en-us/dotnet/api/system.char.parse?view=net-5.0>`_ any of them?
+
+      .. admonition:: Examples
+
+         .. sourcecode:: CSharp
+            :linenos:
+
+            Console.WriteLine("Letters with score 4: {0}", oldPointStructure[4]);
+            
+            foreach(char ch in oldPointStructure[4])
+            {
+                Console.Write(ch + '-');  //dash used to visually separate each char 
+            }
+
+         **Console Output**
+
+         ::
+
+            Letters with score 4: F, H, V, W, Y
+
+            F-,- -H-,- -V-,- -W-,- -Y-
+
+
+C) Scoring Algorithms
+^^^^^^^^^^^^^^^^^^^^^^
+
+We are going to create the individual scoring methods first.  Be sure to test them as you build them.
+Once all three are working, we are going to create a different method to invoke them.
+Read through this section before coding anything.  
+
+Create a *separate* method for each of the following scoring algorithms.
+
+#. **Simple Score:** Define a method that takes a word as a parameter and
+   displays a score. Each letter within the word is worth 1 point.
+#. **Bonus Vowels:** Define a method that takes a word as a parameter and
+   displays a score. Each vowel within the word is worth 3 points, and each
+   consonant is worth 1 point.
+#. **Scrabble Score:** Define a method that takes a word as a parameter and displays a score. 
+   The method uses the *transformed* ``newPointStructure``.
+#. Each method should have a way to display the *original* word and the total value of points.
+   (*Consider:* how many word parameters will you be working with?)
```

Comment:
im not grasping this consider tip. can you add more context? is this asking about how many arguments should be passed into each scoring method?